### PR TITLE
Upgrade the latest macOS version to Monterey for Safari test to test Safari 15

### PR DESCRIPTION
**Pull request by xuesichao**
**Issue #:**

The Sauce Labs releases the support of macOS Monterey recently. https://saucelabs.com/blog/macOS-12-with-safari-15-now-available-on-sauce-labs. We are now able to test our SDK on Safari 15 now. However currently we are testing it on macOS Big Sur with Safari 14.

**Description of changes:**
Update the `latest` version of macOS from `macOS 11.00` to `macOS 12` when the testing browser is Safari.

**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
N/A

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
N/A

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
No

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: integration/js/utils/WebdriverSauceLabs.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const getPlatformName = capabilities => {
   switch (platform) {
     case 'MAC':
       if (browserName === 'safari') {
-        return version === 'latest' ? 'macOS 11.00' : 'macOS 10.13';
+        return version === 'latest' ? 'macOS 12' : 'macOS 10.13';
       }
       return 'macOS 10.14';
```
Comment (ltrung) [inline integration/js/utils/WebdriverSauceLabs.js:14]: can we also update this value to something else oter than 10.14 which is very old?

**Review by jing-chen1 — DISMISSED** [review]

File: integration/js/utils/WebdriverSauceLabs.js (hunk 1 above)
Comment (xuesichao) [reply integration/js/utils/WebdriverSauceLabs.js:14]: I'm not sure which one to use. I think this should be related to the OS we supported.

Comment (xuesichao) [reply integration/js/utils/WebdriverSauceLabs.js]: @ltrung Since the latest version is bumped up for one version, I bumped up this for one as well.

Comment (xuesichao) [reply integration/js/utils/WebdriverSauceLabs.js:14]: Tried to upgrade the default version to `macOS 10.15` or `macOS 11`. But either of them failed to pass the GitHub Action. So have to keep the default Mac version unchanged.

Comment (ltrung) [reply integration/js/utils/WebdriverSauceLabs.js:14]: I think we want to update from macOS 10.14 to BigSur. 10.14 is pretty old :)

**Review by jing-chen1 — APPROVED** [review]

**Review by ltrung — CHANGES_REQUESTED** [review]
Can we update the default mac version too?

**Review by ltrung — APPROVED** [review]

**Pull request merged**